pkg/extensions/tbtc/tbtc.go

```diff
@@ -780,7 +813,39 @@ func (t *tbtc) shouldMonitorDeposit(depositAddress string) bool {
 			depositAddress,
 			err,
 		)
-		return false // return false but don't cache the result in case of error
+		// return false but don't cache the result in case of an error;
+		// it gives a chance to retry the monitoring later but sooner
+		// than notMonitoredDepositsCache caching period.
+		return false
+	}
+
+	hasInitialState, err := utils.ConfirmWithTimeoutDefaultBackoff(
```

Comment:
Why not just increase the `monitoringCachePeriod`?

Reply:
Because this value would depend on subscription settings - specifically the values in `SubscribeOpts` - and blocks mining time in the particular chain/network. The current value of 24 hours is probably enough for Ethereum mainnet and default `SubscribeOpts` fwiw.